Dspace: Add link to parent collection in dmdSec

Hwesta opened the pull request:
Change dmdSec generated for Dspace METS files to contain DublinCore with the identifier and parent collection (isPartOf) instead of a mdRef to the METS.

refs #6273

mistydemeo commented on src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py:
```diff
@@ -254,19 +254,28 @@ def createDublincoreDMDSecFromDBData(type, id):
+        'dc.isPartOf': part_of,
+    }
+    dmdsecs = createDmdSecsFromCSVParsedMetadata(metadata)
+    dmdSec = dmdsecs[0]  # Should only be one dmdSec
```
Worth logging if more than one dmdSec occurs?

Hwesta replied:
I don't think so, since I know exactly what metadata was passed into `createDmdSecsFromCSVParsedMetadata`.

mistydemeo commented:
:thumbsup:

Hwesta committed:
Dspace: Add link to parent collection in dmdSec

refs #6273

Add another dmdSec for Dspace METS files containing Dublin Core with the
identifier and parent collection (isPartOf). Modify calling code to handle
multiple dmdSecs being returned.

Hwesta committed:
DublinCore: Elements namespace as dc or dcterms correctly

DC and dcterms elements are namespaced differently based on their prefix.

Hwesta commented:
Updated to contain both the old mdRef dmdSec and the new identfier dmdSec.

DubinCore metadata is now namespaced differently depending on if it's dc or dcterms.

mistydemeo commented:
Still :+1:

Hwesta committed:
METS, DSpace: Fix METS generation errors

Not all METS files are part of a group, so don't error if one isn't
found. A Dspace METS file may not exist, so pass if one is not found.
New behaviour based on pre-ORM behaviour. Empty file label values may
not be None.

mistydemeo commented:
:+1: on METS change

Hwesta commented on src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py:
```diff
@@ -255,20 +258,48 @@ def createDublincoreDMDSecFromDBData(type, id):
+    dmdsec = etree.Element(ns.metsBNS + "dmdSec", ID=dmdid)
+    dmdsecs[dmdid] = dmdsec
+    xptr_dmdids = [i.get('ID') for i in root.findall("{http://www.loc.gov/METS/}dmdSec")]
+    xptr = "xpointer(id('{}'))".format(' '.join(xptr_dmdids))
```
This will bail if the dmdSec does not contain an ID attribute. Easy fix is `i.get('ID', '')`, but do we want to suppress that error?

mistydemeo replied:
Sounds like we probably want to fail with useful information in that case?

mistydemeo commented:
:thumbsup: to catching and logging the ID error
